repos/spack_repo/builtin/packages/py_rucio_clients/package.py

```diff
@@ -38,24 +41,39 @@ class PyRucioClients(PythonPackage):
     # requirements/requirements.client.txt
     depends_on("python@3.9:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-requests@2.32.2:", type=("build", "run"))
-    depends_on("py-urllib3@1.26.18:", type=("build", "run"))
+    depends_on("py-requests@2.32.2:", type=("build", "run"), when="@:36")
+    depends_on("py-urllib3@1.26.18:", type=("build", "run"), when="@:36")
+    depends_on("py-requests@2.32.3:", type=("build", "run"), when="@37:")
+    depends_on("py-urllib3@2.3.0:", type=("build", "run"), when="@37:")
     depends_on("py-dogpile-cache@1.2.2:", type=("build", "run"))
     depends_on("py-tabulate@0.9.0:", type=("build", "run"))
-    depends_on("py-jsonschema@4.20.0:", type=("build", "run"))
+    depends_on("py-jsonschema@4.20.0:", type=("build", "run"), when="@:36")
+    depends_on("py-jsonschema@4.23.0:", type=("build", "run"), when="@37:")
     depends_on("py-packaging@24.1:", type=("build", "run"), when="@36:")
+    depends_on("py-packaging@24.2:", type=("build", "run"), when="@37:")
     depends_on("py-rich@13.7.1:", type=("build", "run"), when="@36:")
-    depends_on("py-typing-extensions@4.12.2:", type=("build", "run"), when="@36:")
+    depends_on("py-rich@13.9.4:", type=("build", "run"), when="@37:")
+    depends_on("py-typing-extensions@4.12.2:", type=("build", "run"))
+    depends_on("py-click@8.1.7:", type=("build", "run"), when="@37:")
 
     with when("+ssh"):
-        depends_on("py-paramiko@3.4.0:")
+        depends_on("py-paramiko@3.4.0:", when="@36")
```

Comment:
This change results in `py-paramiko` not being required anymore for versions up to 35.
```suggestion
        depends_on("py-paramiko@3.4.0:", when="@:36")
```